app/models/customer.rb

```diff
@@ -2,9 +2,9 @@ class Customer < ActiveRecord::Base
   has_many :invoices
   has_many :transactions, through: :invoices
 
-  def self.favorite_merchant(id)
-    customer = Customer.find(id)
-    merchant_id = customer.invoices.successful_transactions.group_by(&:merchant_id).max_by { |k, v| v.count }.flatten.first
+  def favorite_merchant
+    merchant_id = invoices.successful_transactions.group_by(&:merchant_id)
+      .max_by { |_k, v| v.count }.flatten.first
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.